core/src/main/java/io/confluent/rest/handlers/SniUtils.java

```diff
@@ -28,9 +28,10 @@
 
 public class SniUtils {
   public static String getSniServerName(Request baseRequest) {
-    EndPoint endpoint = baseRequest.getHttpChannel().getEndPoint();
-    if (endpoint instanceof SslConnection.DecryptedEndPoint) {
-      SSLSession session = ((SslConnection.DecryptedEndPoint) endpoint)
+    SslConnection connection = (SslConnection) baseRequest.getConnectionMetaData().getConnection();
+    EndPoint endpoint = connection.getEndPoint();
+    if (endpoint instanceof SslConnection.SslEndPoint) {
+      SSLSession session = ((SslConnection.SslEndPoint) endpoint)
```

Comment:
I think we can just do connection.getSslEndPoint() here, since we already have an `SslConnection`. If we're worried about accounting for non-SSL connections, I think we need to check the type before casing to `SslConnection` on line 31.

Reply:
Also, once we have the `SslConnection`, do we even need to get the `EndPoint`? Looks like we just call `endpoint.getSslConnection()`, can we just do `SSLSession session = connection.getSslEngine().getSession()`?

Reply:
```
SslConnection connection = (SslConnection) baseRequest.getConnectionMetaData().getConnection();
```
this would've failed bc the connection is actually an HttpConnection. I've updated logic for this in a recent commit